server/controllers/auth/signup.js

```diff
@@ -1,3 +1,34 @@
-exports.signup = (req, res) => {
-  res.send(' Hello from signup file');
+const { hash } = require('bcrypt');
+const { jwtSign } = require('../../helpers');
+const { signupSchema } = require('./validation/signup-validation');
+const { users: { getUserByEmailOrUsername, addUser } } = require('../../database/queries');
+
+
+exports.signup = (req, res, next) => {
+  const key = process.env.KEY;
+  const {
+    username, email, password, nativeLangId, learningLangId,
+  } = req.body;
+
+  signupSchema.validate({
+    username, email, password, nativeLangId, learningLangId,
+  })
+    .then(() => getUserByEmailOrUsername(email, username))
+    .then(({ rows }) => {
+      if (rows.length !== 0) throw Error('username or email exists');
+    })
+    .then(() => hash(password, 10))
+    .then((hashed) => addUser({
+      username, email, password: hashed, nativeLangId, learningLangId,
+    }))
+    .then(({ rows }) => jwtSign({ userInfo: { username: rows[0].username, Id: rows[0].id } }, key))
```

Comment:
```suggestion
    .then(({ rows }) => jwtSign({ userInfo: { username: rows[0].username, id: rows[0].id } }, key))
```

be consistent you send it with `id` not `Id` on the login on this PR #86